flytekit/remote/remote.py

```diff
@@ -858,9 +879,17 @@ def register_workflow(
             )
 
         self._resolve_identifier(ResourceType.WORKFLOW, entity.name, version, serialization_settings)
-        ident = asyncio.run(
-            self._serialize_and_register(entity, serialization_settings, version, options, default_launch_plan)
-        )
+        try:
+            import nest_asyncio
+
+            nest_asyncio.apply()
+            ident = asyncio.run(
```

Comment:
Is this because the remote is trying to use Jupyter's event loop? If so, can `FlyteRemote` create it's own event loop to process these async calls?

Reply:
Actually, can we just use the running loop?

```
def get_working_loop():
    """Returns a running event loop."""
    try:
        return asyncio.get_running_loop()
    except RuntimeError:
        with warnings.catch_warnings():
            warnings.simplefilter("error", DeprecationWarning)
            try:
                return asyncio.get_event_loop_policy().get_event_loop()
            except DeprecationWarning:
                return asyncio.get_event_loop_policy().new_event_loop()
```

Reply:
@thomasjpfan I think the main problem here is that if we call `asyncio.run` directly. There will have nested asyncio event loops exist (the jupyter one and the `asyncio.run` one). This is not permitted in asyncio package, so we need to use nest_asyncio package to resolve this problem. 

Honestly, I thought that the try & except here is not necessary as `import asyncio` should resolve the problem already.

Reply:
With `get_working_loop`, we never run `asyncio.run`. The pattern would be:

```python
loop = get_working_loop()
loop.run(...)
```

Where `get_working_loop` will either create a loop if there is not one running or return the Jupyter event loop. This way, there is no more nested event loops.

(Note I have not tested this out.)

Reply:
I got it, but there is another problem. We couldn't run a new task in a running event loop, at least the function `loop.run_until_complete()` will throw error since Jupyter event loop is always running.
I also tried `create_task()` function, but not yet figured out how to get the result without changing the function into asynchronous function.

Reply:
@thomasjpfan I couldn't figure out the approach without the `nest_asyncio` package, and I also think we can open another PR to fix this issue. 
As for now, I was thinking of removing the `nest_asyncio` package and asking the user to add this manually 
```
import nest_asyncio
nest_asyncio.apply()
```
at the beginning of their Jupyter Notebook if they want to use it in the Jupyter Notebook environment so that we don't need to add an extra dependency.

Reply:
I'm undecided on this especially because https://github.com/erdewit/nest_asyncio is unmaintained. 

If we can not get this working with Jupyter's event loop, then I'll like to run a flytekit owned event loop on another thread: https://github.com/flyteorg/flytekit/pull/2744

I really do not like mixing async code with sync code. @pingsutw What do you think of having a sync version of `_serialize_and_register` that runs in jupyter? Another options is to revert https://github.com/flyteorg/flytekit/pull/2276.

Reply:
@thomasjpfan Should we include the Flytekit-owned event loop in this PR? For now, it might be better to ask the user to manually add `nest_asyncio`, as shown in the example in this PR’s description, instead of including it in this project. We could then create a separate PR to address this issue.

I’m almost finished with this PR, except for adding a few unit tests. Could you please review the code again and let me know if there’s anything else that needs attention or discussion? Thanks!